test/gurka/gurka.cc

```diff
@@ -507,7 +507,7 @@ findEdge(valhalla::baldr::GraphReader& reader,
             auto forward_edge_id = tile_id;
             forward_edge_id.set_id(i);
             auto reverse_edge_id = tile->GetOpposingEdgeId(forward_directed_edge);
-            auto* reverse_directed_edge = tile->directededge(i);
+            auto* reverse_directed_edge = tile->directededge(reverse_edge_id.id());
```

Comment:
yep you found a bug here, i have no clue how the hell this is possible to have been here so long but anyway...

i would suggest we change this whole bit to be more like:

```c++
const DirectedEdge* reverse_directed_edge = nullptr;
graph_tile_ptr reverse_tile = tile;
auto reverse_edge_id = GetOpposingEdgeId(forward_edge_id, reverse_directed_edge, reverse_tile);
```

Edges are stored in teh tiles where they originate, that is where their begin node is. For example: imagine we have 2 tiles that are adjacent call them A and B. If forward edge here originates in A and ends in B then its opposing edge will originate in B and end in A. That means that forward edge is sorted in A and reverse edge is stored in B. One cannot ask tile A for reverse edge and get the right answer. Now the fix you did would actually fix this crazy bug in the case that forward edge and reverse edge were in the same tile, so its definitely better, but its stil not right. The above snippet I posted should do the trick and im guessing other unit test fall out will ensue...

Reply:
it looks like it only affected that time tracking test regardless of the fix so yeah i think you can just make the change i suggest here and we should be good to go on that front!

Reply:
Thanks for the explanation, I'm on it.